packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -94,6 +99,10 @@ function* getContentInfo({ contentId, contentType }: GetPurchaseConfigArgs) {
   const trackInfo = yield* select(getTrack, { id: contentId })
   const purchaseConditions =
     trackInfo?.stream_conditions ?? trackInfo?.download_conditions
+  const purchaseAccess =
+    trackInfo?.is_download_gated && !trackInfo?.is_stream_gated
```

Comment:
I'm a little confused by this: If it's download gated and _not_ stream gated, we purchase download access, otherwise we purchase stream access?
Wouldn't we always want to purchase download access if it's download gated?

Reply:
the way the code is written makes sense to me, otherwise we'd always be purchasing download access.

Reply:
although, i'd write it a bit differently. `trackInfo?.is_download_gated` is never `false` if we are purchasing access anyway.

I'd write it as `trackInfo?.is_stream_gated ? PurchaseAccess.STREAM : PurchaseAccess.DOWNLOAD`

Reply:
will add a comment